internal/preflight/preflight.go

```diff
@@ -55,13 +62,47 @@ func (p *preflight) PreflightUpgradeCheck(ctx context.Context, mod *kmmv1beta1.M
 		return false, fmt.Sprintf("Failed to substitute template in kernel mapping in the module %s for kernel version %s", mod.Name, kernelVersion)
 	}
 
-	return p.verifyImage(ctx, mapping, mod, kernelVersion)
+	err = p.statusUpdater.PreflightSetVerificationStage(ctx, pv, mod.Name, kmmv1beta1.VerificationStageImage)
+	if err != nil {
+		log.Info(utils.WarnString("failed to update the stage of Module CR in preflight to image stage"), "module", mod.Name, "error", err)
+	}
+
+	res, msg := p.helper.verifyImage(ctx, mapping, mod, kernelVersion)
+	if res || (mapping.Build == nil && mod.Spec.ModuleLoader.Container.Build == nil) {
+		return res, msg
+	}
```

Comment:
Took me like 5 min to understand that if so I will suggest:
```suggestion
	if exists, msg := p.helper.verifyImage(ctx, mapping, mod, kernelVersion); exists {
	        return true, msg
	 else if mapping.Build == nil && mod.Spec.ModuleLoader.Container.Build == nil {
		return false, msg
	}
```

Reply:
i am not sure that using if .. else if.. is more readable, what if i change the res to imageVerified?

Reply:
> what if i change the res to imageVerified?

This will be great.

You can keep it if you want and I won't object to it. Just didn't understood that the condition for a few minutes.
This is how I understood it (wan't intuitive for me):
1. if the image was verified then we should return `true` with the correct message
2. if we couldn't verify the image, we want to try to build it but if there is not build option then we have to return `false` and the relevant message.

I found it very unclear the way it was written without the clear separation of the cases but if you prefer it that way I will merge it as you like.

Reply:
please take a look how if it is more readable now, with the changed variable name

Reply:
It is better, still a bit confusing to me but let's merge that way.